x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -3348,6 +3348,40 @@ public void testBucketFailsOnFieldArgument() {
         );
     }
 
+    /*
+     * Project[[bucket(salary, 1000.) + 1{r}#3, bucket(salary, 1000.){r}#5]]
+        \_Eval[[bucket(salary, 1000.){r}#5 + 1[INTEGER] AS bucket(salary, 1000.) + 1]]
+          \_Limit[1000[INTEGER]]
+            \_Aggregate[[bucket(salary, 1000.){r}#5],[bucket(salary, 1000.){r}#5]]
+              \_Eval[[BUCKET(salary{f}#12,1000.0[DOUBLE]) AS bucket(salary, 1000.)]]
+                \_EsRelation[test][_meta_field{f}#13, emp_no{f}#7, first_name{f}#8, ge..]
+     */
+    public void testBucketWithAggExpression() {
+        var plan = plan("""
+            from test
+            | stats bucket(salary, 1000.) + 1 by bucket(salary, 1000.)
+            """);
+        var project = as(plan, Project.class);
+        var evalTop = as(project.child(), Eval.class);
+        var limit = as(evalTop.child(), Limit.class);
+        var agg = as(limit.child(), Aggregate.class);
+        var evalBottom = as(agg.child(), Eval.class);
+        var relation = as(evalBottom.child(), EsRelation.class);
+
+        assertThat(evalTop.fields().size(), is(1));
+        assertThat(evalTop.fields().get(0), instanceOf(Alias.class));
+        assertThat(evalTop.fields().get(0).child(), instanceOf(Add.class));
+        var add = (Add) evalTop.fields().get(0).child();
+        assertThat(add.left(), instanceOf(ReferenceAttribute.class));
+        var ref = (ReferenceAttribute) add.left();
+
+        assertThat(evalBottom.fields().size(), is(1));
+        assertThat(evalBottom.fields().get(0), instanceOf(Alias.class));
+        var alias = evalBottom.fields().get(0);
+
+        assertEquals(ref, alias.toAttribute());
+    }
```

Comment:
Nit: since this change updates the groupings, I think we should check that they are correct in this test.

Reply:
> this change updates the groupings

The groupings aren't updated by this change, they "remain" updated by the rule as it was (i.e. taken out into a distinct `EVAL`). But let me know if I understood your note wrongly.

Reply:
Sorry, you are correct - this change updates the aggregates, not the groupings.

Anyway, what I meant is: we don't make assertions about `agg` at all, we only check that there's an aggregate at all. I think the test is a bit incomplete if we don't also check that the updated aggs correctly reference the synthetic eval that was added before `agg`.

Reply:
I see. Still, this bit should be tested by those tests added along the rule (like `testNestedExpressionsInBothAggsAndGroups()`), but I guess more checks are generally good, so I've added more.